web-components/src/components/audio-player/AudioPlayer.ts

```diff
@@ -0,0 +1,434 @@
+  },
+  {
+    value: 0.75,
+    label: "0.75"
+  },
+  {
+    value: 1,
+    label: "Normal"
+  },
+  {
+    value: 1.25,
+    label: "1.25"
+  },
+  {
+    value: 1.5,
+    label: "1.50"
+  },
+  {
+    value: 1.75,
+    label: "1.75"
+  },
+  {
+    value: 2.0,
+    label: "2"
+  }
+];
+
+export type Label = {
+  ariaLabel: string;
+  tooltipText: string;
+};
+
+export type LabelMap = {
+  playBtn: Label;
+  duration: Label;
+  timeline: Label;
+  volumeBtn: Label;
+  volumeSlider: Label;
+  playbackSpeedBtn: Label;
+};
+
+export const defaultLabelMap = {
+  playBtn: {
+    ariaLabel: "Play / Pause",
+    tooltipText: "Play / Pause"
+  },
+  duration: {
+    ariaLabel: "Duration:",
+    tooltipText: "Duration"
+  },
+  timeline: {
+    ariaLabel: "Audio Seekbar",
+    tooltipText: "Audio Seekbar"
+  },
+  volumeBtn: {
+    ariaLabel: "Mute / Unmute Audio",
+    tooltipText: "Mute / Unmute"
+  },
+  volumeSlider: {
+    ariaLabel: "Volume level",
+    tooltipText: "Volume level"
+  },
+  playbackSpeedBtn: {
+    ariaLabel: "Playback speed",
+    tooltipText: "Playback speed"
+  }
+};
+
+export namespace AudioPlayer {
+  @customElementWithCheck("md-audio-player")
+  export class ELEMENT extends LitElement {
+    @property({ type: String }) src = "";
+    @property({ type: Object }) labelMap: LabelMap;
+    @internalProperty() events: AudioEvent[] = [];
+    @internalProperty() audio: HTMLAudioElement;
+    @internalProperty() isPlaying = false;
+    @internalProperty() isMuted = false;
+    @internalProperty() duration = 0;
+    @internalProperty() volume = 100;
+    @internalProperty() currentTime = 0;
+    @internalProperty() showSpeedPopup = false;
+    @internalProperty() playbackSpeed = 1;
+    @internalProperty() volumeExpanded = false;
+
+    constructor() {
+      super();
+      this.setAudioEventListeners();
+      this.audio = new Audio();
+      this.labelMap = defaultLabelMap;
+    }
+
+    firstUpdated(changedProperties: PropertyValues) {
+      super.firstUpdated(changedProperties);
+      if (changedProperties.has("src")) {
+        this.audio.src = this.src;
+      }
+    }
+
+    updated(changedProperties: PropertyValues) {
+      super.updated(changedProperties);
+      if (changedProperties.has("src")) {
+        this.audio.src = this.src;
+      }
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+      this.requestUpdate("src");
+      document.addEventListener("click", this.onOutsideClick);
+      document.addEventListener("keydown", this.handleKeyDown);
+
+      for (const event of this.events) {
+        this.audio.addEventListener(event.name, event.onEvent, false);
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      document.removeEventListener("click", this.onOutsideClick);
+      document.removeEventListener("keydown", this.handleKeyDown);
+
+      for (const event of this.events) {
+        this.audio.removeEventListener(event.name, event.onEvent, false);
```

Comment:
removeEventListener doesn't work for anonymous function. It needs the same function reference. Refer this:
https://stackoverflow.com/questions/65379045/remove-event-listener-with-an-anonymous-function-with-a-local-scope-variable

Reply:
switched these out for event property handlers. no need to add or remove listeners this way.